File: CPP/3-TwoPointers/2-two_sum_2.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> twoSum(vector<int>& numbers, int target) {
    int index1 = 0;
    int index2 = numbers.size() - 1;

    while (index1 < index2) {
        int sum = numbers[index1] + numbers[index2];

        if (sum == target) {
            return {index1 + 1, index2 + 1};
        }

        if (sum < target) {
            index1++;
        } else {
            index2--;
        }
    }
    return {};
}

int main() {
    vector<int> numbers = {5, 25, 75};
    int target = 100;

    vector<int> resp = twoSum(numbers, target);

    for (int& num : resp) {
        cout << num << " ";
    }

    return 0;
}

File: PYTHON/3-TwoPointers/2-two_sum_2.py
from typing import List

def twoSum(numbers: List[int], target: int) -> List[int]:
    esq, dir = 0, len(numbers) - 1

    while esq < dir:
        soma = numbers[esq] + numbers[dir]
        
        if soma < target:
            esq += 1
        elif soma > target:
            dir -= 1
        else:
            return [esq + 1, dir + 1]
    
    return []

if __name__ == "__main__":
    numbers = [1, 3, 5, 7]
    target = 4
    print(twoSum(numbers, target))